src/classes/AllocationParent.cls

```diff
@@ -0,0 +1,174 @@
+        record = getRecord();
+        setPropertiesFromDescribe();
+
+        if (UserInfo.isMultiCurrencyOrganization()) {
+            currencyCode = (String)record.get(UTIL_Currency.CURRENCY_ISO_CODE_FIELD);
+        }
+    }
+
+    public Id getId() {
+        return id;
+    }
+
+    public SObjectType getSObjectType() {
+        return objectType;
+    }
+
+    public String getKeyPrefix() {
+        return keyPrefix;
+    }
+
+    public String getObjectLabelPlural() {
+        return objectLabelPlural;
+    }
+
+    public String getObjectApiName() {
+        return objectApiName;
+    }
+
+    public String getRecordName() {
+        return (String)record?.get('Name');
+    }
+
+    public String getCurrencyCode() {
+        return currencyCode;
+    }
+
+    public String getAllocationLookupApiName() {
+        return allocationLookupApiNameBySObjectType.get(objectType);
+    }
+
+    public Boolean isAnOpportunity() {
+        return objectType == Opportunity.SObjectType;
+    }
+
+    public Boolean isAPayment() {
+        return objectType == npe01__OppPayment__c.SObjectType;
+    }
+
+    public Boolean hasAnAmountField() {
+        return isAnOpportunity() || isAPayment();
+    }
+
+    public Boolean hasAnAmountGreaterThanZero() {
+        return hasAnAmountField() && getAmount() > 0;
+    }
+
+    public Decimal getAmount() {
+        if (donationAmount == 0 || donationAmount == null) {
+
+            Boolean shouldBeAmountFromOpportunity = isAnOpportunity() && record?.get(OPPORTUNITY_AMOUNT) != null;
+            if (shouldBeAmountFromOpportunity) {
+                donationAmount = (Decimal)record.get(OPPORTUNITY_AMOUNT);
+            }
+
+            Boolean shouldBeAmountFromPayment = isAPayment() && record?.get(PAYMENT_AMOUNT) != null;
+            if (shouldBeAmountFromPayment) {
+                donationAmount = (Decimal)record.get(PAYMENT_AMOUNT);
+            }
+        }
+
+        return donationAmount;
+    }
+
+    private void setPropertiesFromDescribe() {
+        DescribeSObjectResult describe = objectType.getDescribe();
+        keyPrefix = describe.getKeyPrefix();
+        objectLabelPlural = describe.getLabelPlural();
+        objectApiName = describe.getName();
+    }
+
+    private SObject getRecord() {
+        Set<String> fields = getSelectFieldsFor(objectType);
+
+        UTIL_Query soql = new UTIL_Query();
+        String queryString = soql
+            .withFrom(objectType)
+            .withSelectFields(fields)
+            .withWhere('Id = :id')
+            .withLimit(1)
+            .build();
+
+        return Database.query(queryString);
+    }
+
+    private Set<String> getSelectFieldsFor(SObjectType objectType) {
+        Set<String> selectFields = new Set<String>{'Id', 'Name'};
+        List<String> additionalFields = selectFieldsBySObjectType.get(objectType);
+        if (additionalFields != null) {
+            selectFields.addAll(selectFieldsBySObjectType.get(objectType));
```

Comment:
instead of performing an additional get() on the key here, can you just add `additionalFields` in the addAll()?

Reply:
🤦‍♂️ updated!